pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -133,13 +137,27 @@ func (inferredSpan *InferredSpan) CompleteInferredSpan(
 	})
 }
 
+// GenerateSpanId creates a secure random span id in specific scenarios, otherwise return a pseudo random id
+func GenerateSpanId() uint64 {
+	isSnapStart := os.Getenv(tags.InitType) == tags.SnapStartValue
+	if isSnapStart {
+		max := big.NewInt(math.MaxInt64)
```

Comment:
When snap starting this block of code will only produce an id with 63 bits of randomness, MaxInt64 is 2 ^ 63, where as MaxUint64 is 2 ^ 64. However the return outside this block with random.Random.Uint64() will produce 64 bits of randomness. Not sure if that was your intention, since we often limit our trace-ids to 63bits of randomness for compatibility with certain tracers.